instrumentation/logback/logback-appender-1.0/library/src/main/java/io/opentelemetry/instrumentation/logback/appender/v1_0/internal/LoggingEventMapper.java

```diff
@@ -639,27 +639,43 @@ private static boolean supportsLogstashStructuredArguments() {
   @NoMuzzle
   private void captureLogstashStructuredArguments(LogRecordBuilder builder, Object[] arguments) {
     for (Object argument : arguments) {
-      if (isLogstashStructuredArgument(argument)) {
-        captureLogstashStructuredArgument(builder, argument);
+      if (isLogstashSingleStructuredArgument(argument)) {
+        captureLogstashSingleStructuredArgument(builder, argument);
+      }
+      if (isLogstashMapStructuredArgument(argument)) {
+        captureLogstashMapStructuredArgument(builder, argument);
       }
     }
   }
 
   @NoMuzzle
-  private static boolean isLogstashStructuredArgument(Object argument) {
+  private static boolean isLogstashSingleStructuredArgument(Object argument) {
     // StructuredArguments implement the marker interface, so we can check for it
     // without importing the class directly (which may not be available at runtime)
     return argument instanceof SingleFieldAppendingMarker;
   }
 
   @NoMuzzle
-  private void captureLogstashStructuredArgument(LogRecordBuilder builder, Object argument) {
+  private void captureLogstashSingleStructuredArgument(LogRecordBuilder builder, Object argument) {
     // StructuredArguments created by v() or keyValue() extend SingleFieldAppendingMarker
     // which has getFieldName() and provides field value via reflection
     SingleFieldAppendingMarker marker = (SingleFieldAppendingMarker) argument;
     captureLogstashMarker(builder, marker);
   }
```

Comment:
I'd delete this method and use `captureLogstashMarker(builder, marker);` instead. Casting here doesn't seem to be useful.